%% NOTE -- ONLY EDIT THE .Rnw FILE!!!  The .tex file is
% likely to be overwritten.
%
%\VignetteIndexEntry{HDMT}
%\VignetteDepends{}
%\VignetteKeywords{Documentation}
%\VignettePackage{HDMT}
\documentclass[12pt]{article}
\usepackage{times}
\usepackage{hyperref}
%\usepackage[authoryear,round]{natbib}

\textwidth=6.2in
\textheight=8.5in
%\parskip=.3cm
\oddsidemargin=.1in
\evensidemargin=.1in
\headheight=-.3in

\newcommand{\scscst}{\scriptscriptstyle}
\newcommand{\scst}{\scriptstyle}

\newcommand{\R}{{\textsf{R}}}
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\term}[1]{{\emph{#1}}}
\newcommand{\Rpackage}[1]{\textsf{#1}}
\newcommand{\Rfunction}[1]{\texttt{#1}}
\newcommand{\Robject}[1]{\texttt{#1}}
\newcommand{\Rclass}[1]{{\textit{#1}}}
\newcommand{\Rmethod}[1]{{\textit{#1}}}
\newcommand{\Rfunarg}[1]{{\textit{#1}}}

%\bibliographystyle{plainnat}

\begin{document}
\SweaveOpts{concordance=TRUE}

\title{The \Rpackage{'HDMT'} package: A Multiple Testing Procedure For High-dimensional Mediation hypotheses}
\author{Xiaoyu Wang, James Y. Dai}
\maketitle

\section{Introduction}

Mediation analysis is of rising interest in clinical trials and epidemiology. The advance of high-throughput technologies has made it possible to interrogate molecular phenotypes such as gene expression and DNA methylation in a genome-wide fashion, some of which may act as intermediaries of treatment, external exposures and life-style risk factors in the etiological pathway to diseases or traits. When testing for mediation in high-dimensional studies like ours \cite{Dai2020}, properly controlling the type I error rate remains a challenge due to the composite null hypothesis. Among existing methods, the joint significance (JS) test is an intersection-union test using the maximum p-value for testing the two parameters, though a naive significance rule based on the uniform null p-value distribution (JS-uniform) may yield an overly conservative type I error rate and therefore low power. This is particularly a concern for high-dimensional mediation hypotheses for genome-wide molecular intermediaries such as DNA methylation. In this R package we develop a multiple-testing procedure that accurately controls the family-wise error rate (FWER) and the false discovery rate (FDR) for testing high-dimensional mediation composite null hypotheses. The core of our procedure is based on estimating the proportions of three types of component null hypotheses and deriving the corresponding mixture distribution (JS-mixture) of null p-values. Theoretical derivation and extensive simulations show that the proposed procedure provides adequate control of FWER and FDR when the number of mediation hypotheses is large.

<<loadLibrary, echo=FALSE, results=hide>>=

library(HDMT)
@

\section{Examples}

We show two examples assessing the mediation role of DNA methylation in prostate cancer studies. The first study is on assessing the mediation potential role of DNA methylation in genetic regulation of gene expression in primary prostate cancer (PCa) samples from The Cancer Genome Atlas (TCGA). The second study investigated the potential mediation of DNA mColumn 1 contains the p-values for testing if an exposure is associated with the mediator ($\alpha\neq0$). Column 2 contains the p-value for testing if a mediator is associated with the outcome after adjusted for the exposure ($\beta\neq0$) methylation in exercise effect on prostate cancer progression in a Seattle-based patient cohort.
<<data>>=
data(snp_input)
dim(snp_input)
data(exercise_input)
dim(exercise_input)
@
The input data matrix contains two columns of p-values for candidate mediators.  See notation in \cite{Dai2020}. 

\subsection{DNAMethylation in Genetic Regulation of Gene Expression Among 147 Prostate Cancer Risk SNPs}
We first read the input data matrix first. To save time for compiling this vignettes file, we use 10\% of p-values in the input data matrix.  To reproduce the figure in the paper, one need to run the code on the entire data matrix.
<<input_pvalues>>=
input_pvalues <- snp_input

input_pvalues=input_pvalues[sample(1:nrow(input_pvalues),
                  size=ceiling(nrow(input_pvalues)/10)),]
@
The first step of the procedure is to estimate the proportions of the three type of null hypotheses
<<nullprop>>=
nullprop <- null_estimation(input_pvalues)
nullprop
@
We next compute the expected quantiles of the mixture null distribution of $p_{max}$ (maximum of two p-values)
using either the approximation (\texttt{exact}=0) method or the exact method (\texttt{exact}=1). This set of quantiles can be used to draw the corrected q-q plot. 
<<pnull1>>=
pnull1<-adjust_quantile(nullprop$alpha00,nullprop$alpha01,nullprop$alpha10,
                        nullprop$alpha1,nullprop$alpha2,input_pvalues,exact=1)
@
'HDMT' provides a function correct\_qqplot to draw the corrected quantile-quantile plot for 
$p_{max}$, based on the estimated quantile of the mixture null distribution (green dots) and compared to the standard q-q plot based on the uniform distribution (red dots). 
<<plot_snp,fig=TRUE>>=
pmax <- apply(input_pvalues,1,max)
correct_qqplot(pmax, pnull1)
@


We can also compute the pointwise estimated FDR for $p_{max}$ using either the approximation method or the exact method.
<<fdr>>=
efdr <- fdr_est(nullprop$alpha00,nullprop$alpha01,nullprop$alpha10,
                nullprop$alpha1,nullprop$alpha2,input_pvalues,exact=0)

plot(pmax[order(pmax)],efdr[order(pmax)],type="l",ylim=c(0,1),
     xlab="p-max",ylab="Estimated FDR")
@

\subsection{DNA Methylation and the Association of Vigorous Physical Activity With Lower Risk of Metastatic Progression}

For this example, we only included 10\% of p-values from the genome-wide testing as shown in the paper, due to data storage space limit. We read in the input data matrix with two columns of p-values as follows. 
<<input_pvalues>>=
input_pvalues <- exercise_input
#To save time, we use 10% of rows
input_pvalues=input_pvalues[sample(1:nrow(input_pvalues),
                  size=ceiling(nrow(input_pvalues)/10)),]
@
The estimation procedures are identical to the previous example:
<<nullprop>>=
nullprop <- null_estimation(input_pvalues)
@
We can compute the null distribution of pmax using the approximation method in Section 2.2, and we can also compute the null distribution of pmax using the exact method in Section 2.4.
<<pnull>>=
pnull<-adjust_quantile(nullprop$alpha00,nullprop$alpha01,nullprop$alpha10,
nullprop$alpha1,nullprop$alpha2,input_pvalues,exact=0)
@
<<pnull1>>=
pnull1<-adjust_quantile(nullprop$alpha00,nullprop$alpha01,nullprop$alpha10,
nullprop$alpha1,nullprop$alpha2,input_pvalues,exact=1)
@
The Q-Q plot based on the approximation method is shown as follows:
<<plot_excercise1,fig=TRUE>>=
pmax <- apply(input_pvalues,1,max)
correct_qqplot(pmax, pnull)
@

\vspace*{2\baselineskip}
The Q-Q plot based on the exact method is shown as follows:
<<plot_excercise2,fig=TRUE>>=
correct_qqplot(pmax, pnull1)
@


\section{session information}

The version number of \R{} and packages loaded for generating the vignette were:

\begin{verbatim}
<<sessionInfo, echo=FALSE, results=tex>>=
sessionInfo()
@
\end{verbatim}

\bibliographystyle{plain}
\bibliography{HDMT}
 
\end{document}

